alf/networks/networks.py

```diff
@@ -308,3 +315,186 @@ def forward(self, input, state):
+            state_spec=(alf.TensorSpec((input_size, )),
+                        alf.TensorSpec((output_size, )))),
+        self._input_size = input_size
+        self._output_size = output_size
+        self._activation = activation
+        self._std_init = std_init
+        self._weight = nn.Parameter(torch.empty(output_size, input_size))
+        self._weight_sigma = nn.Parameter(torch.empty(output_size, input_size))
+        self._bias = nn.Parameter(torch.empty(output_size))
+        self._bias_sigma = nn.Parameter(torch.empty(output_size))
+        self._use_bn = use_bn
+        self._use_ln = use_ln
+        if use_bn:
+            self._bn = bn_ctor(output_size)
+        else:
+            self._bn = None
+        if use_ln:
+            self._ln = nn.LayerNorm(output_size)
+        else:
+            self._ln = None
+        self._new_noise_prob = new_noise_prob
+        self._kernel_initializer = kernel_initializer
+        self._kernel_init_gain = kernel_init_gain
+        self._bias_init_value = bias_init_value
+        self._bias_initializer = bias_initializer
+        self.reset_parameters()
+        if weight_opt_args:
+            self._weight.opt_args = weight_opt_args
+            weight_opt_args = copy.copy(weight_opt_args)
+            weight_opt_args['zero_mean'] = False
+            weight_opt_args['fixed_norm'] = False
+            self._weight_sigma.opt_args = weight_opt_args
+        if bias_opt_args and self._bias is not None:
+            self._bias.opt_args = bias_opt_args
+            bias_opt_args = copy.copy(bias_opt_args)
+            bias_opt_args['zero_mean'] = False
+            self._bias_sigma.opt_args = bias_opt_args
+
+    @property
+    def input_size(self):
+        return self._input_size
+
+    @property
+    def output_size(self):
+        return self._output_size
+
+    @property
+    def weight(self):
+        return self._weight
+
+    @property
+    def bias(self):
+        return self._bias
+
+    def reset_parameters(self):
+        """Initialize the parameters."""
+        if self._kernel_initializer is None:
+            variance_scaling_init(
+                self._weight.data,
+                gain=self._kernel_init_gain,
+                nonlinearity=self._activation)
+        else:
+            self._kernel_initializer(self._weight.data)
+        self._weight_sigma.data.fill_(
+            self._std_init / math.sqrt(self._input_size))
+        if self._bias_initializer is not None:
+            self._bias_initializer(self._bias.data)
+        else:
+            nn.init.constant_(self._bias.data, self._bias_init_value)
+        self._bias_sigma.data.fill_(
+            self._std_init / math.sqrt(self._output_size))
+        if self._use_ln:
+            self._ln.reset_parameters()
+        if self._use_bn:
+            self._bn.reset_parameters()
+
+    def _scale_noise(self, size):
+        x = torch.randn(size)
+        return x.sign().mul_(x.abs().sqrt_())
```

Comment:
This is no longer Gaussian? Why not just `return torch.randn(size)`?

Reply:
This follows the implementation described in the cited paper.